full-stack-tests/presentation/src/learning-snippets/Content.test.ts

```diff
@@ -11,582 +11,1002 @@ import { getFieldByLabel, tryGetFieldByLabel } from "../Utils";
+              priority: 1,
+            }],
+          }],
+        };
+        // __PUBLISH_EXTRACT_END__
+
+        // Ensure that only `bis.ViewDefinition` instances are selected.
+        const content = await Presentation.presentation.getContent({
+          imodel,
+          rulesetOrId: ruleset,
+          keys: new KeySet(),
+          descriptor: {},
+        });
+
+        expect(content!.contentSet.length).to.eq(2);
+        const calculatedField = tryGetFieldByLabel(content!.descriptor.fields, "Modeled Element");
+        expect(content!.contentSet[0].displayValues[calculatedField!.name]).to.eq("BisCore.DictionaryModel");
+        expect(content!.contentSet[1].displayValues[calculatedField!.name]).to.eq("Properties_60InstancesWithUrl2");
+      });
+
+      it("uses `relatedProperties` attribute", async () => {
+        // __PUBLISH_EXTRACT_START__ ContentInstancesOfSpecificClasses.relatedProperties.Ruleset
+        // The specification returns content for `bis.SpatialViewDefinition` and
+        // related `bis.DisplayStyle` properties.
+        const ruleset: Ruleset = {
+          id: "example",
+          rules: [{
+            ruleType: RuleTypes.Content,
+            specifications: [{
+              specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: ["SpatialViewDefinition"] },
+              relatedProperties: [{
+                propertiesSource: {
+                  relationship: { schemaName: "BisCore", className: "ViewDefinitionUsesDisplayStyle" },
+                  direction: RelationshipDirection.Forward,
+                },
+              }],
+            }],
           }],
-          propertyOverrides: [{
-            name: "*",
-            categoryId: "custom-category",
+        };
+        // __PUBLISH_EXTRACT_END__
+
+        // Ensure that derived `bis.DisplayStyle` instance properties are also returned with `bis.SpatialViewDefinition` content.
+        const content = await Presentation.presentation.getContent({
+          imodel,
+          rulesetOrId: ruleset,
+          keys: new KeySet(),
+          descriptor: {},
+        });
+        expect(content!.descriptor.fields).to.containSubset([
+          { label: "3D Display Style" },
+          { label: "Category Selector" },
+          { label: "Code" },
+          { label: "Description" },
+          { label: "Display Style" },
+          { label: "Extents" },
+          { label: "Eye Point" },
+          { label: "Focus Distance" },
+          { label: "Is Camera On" },
+          { label: "Is Private" },
+          { label: "Lens Angle" },
+          { label: "Model" },
+          { label: "Model Selector" },
+          { label: "Origin" },
+          { label: "Pitch" },
+          { label: "Roll" },
+          { label: "User Label" },
+          { label: "Yaw" },
+        ]).and.to.have.lengthOf(18);
```

Comment:
I think it's more important to check properties of the related instance rather than properties of the content instance.

Reply:
Considering that there is only 1 related property, would you keep the length check for all fields? It's an easy way to check that the specific class properties are also returned, but if that's not what we want to test, then it would probably make sense to remove it too.

Reply:
Let's keep the length check. And then check that there's the "3D Display Style" field with correct nested fields.